solutions/CF Party and Sweets.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: cf-1158A
+source: CF
+title: The Party and Sweets
+author: Jesse Choe
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/66993)
+
+**Time Complexity:** $\mathcal{O}(N\log N + M\log M)$
+
+The problem asks us to find the minimum sum of sweets which a boy could give to
+a girl with the given constraints on the minimum number of sweets a boy could
+give to a girl and the maximum number of sweets a girl has received from a boy.
+Additionally, the problem asks us to print $-1$ if any given arrangement of
+sweets does not follow the given constraints.
+
+Let's consider the following example from the sample input, which outputs $-1$:
+
+```
+2 2
+0 1
+1 0
+```
+
+The answer to this example is $-1$ since boy $2$ ends up giving girl $2$ too
+many sweets, even if boy $2$ gives girl $2$ the minimum number of sweets he
+could possibly give, which is $1$. However, girl $2$ received a maximum of no
+sweets, making it impossible.
+
+As we can see in this example, if a single boy gives more sweets than a girl
+received, then any arrangement of sweets is impossible with the constraints.
+More formally, if the
+$max(b_1, b_2, \dots, b_n-1, b_n) > min(g_1, g_2, \dots, g_m-1, g_m)$, then the
+answer is $-1$.
+
+If the input provided has an arrrangement of sweets that follows the given
+constraints, then we can greedily search for the answer.
+
+Since our objective is to minimize the total sum of sweets, it is optimal to
+make each sweet in the arrangement as small as possible. Since the smallest
+number of sweet a boy $i$ could give to a girl $j$ is $b_i$, we could "give"
+each girl $b_i$ sweets, which is: $\sum_{i=1}^{n} b_i \times m$.
```

Comment:
delete everything before the equation and reword

Reply:
alright